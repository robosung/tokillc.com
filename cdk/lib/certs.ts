import {
    Stack,
    StackProps,
    aws_certificatemanager as acm,
    aws_route53 as r53,
    aws_ssm as ssm,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';

export class Certs extends Stack {
    constructor(scope: Construct, id: string, props: StackProps,) {
        super(scope, id);
        // Cert
        // Top level
        const topZone = r53.PublicHostedZone.fromHostedZoneAttributes(this, 'TopZone', {
            hostedZoneId: ssm.StringParameter.valueForStringParameter(this, 'tokillc-hostedzoneid'),
            zoneName: ssm.StringParameter.valueForStringParameter(this, 'tokillc-hostedzonename'),
        });


        const accountCert = new acm.Certificate(this, `tokillc-certificate`, {
            domainName: `*.${topZone.zoneName}`,
            validation: acm.CertificateValidation.fromDns(topZone),
        });


        new ssm.StringParameter(this, 'tokillc-wildcardcertarn', {
            description: 'Hosted Zone Cert',
            parameterName: 'tokillc-wildcardcertarn',
            stringValue: accountCert.certificateArn,
        });
    }
}