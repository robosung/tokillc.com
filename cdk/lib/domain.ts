import {
    Stack,
    StackProps,
    aws_certificatemanager as acm,
    aws_route53 as r53,
    aws_ssm as ssm,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';

export class Domain extends Stack {
    constructor(scope: Construct, id: string, props: StackProps,) {
        super(scope, id);

        //Domain
        const zoneName = `tokillc.com`;
        const domainZone = new r53.PublicHostedZone(this, `${zoneName}-domain`, { zoneName: zoneName });

        new ssm.StringParameter(this, 'tokillc-hostedzoneid', {
            description: 'Hosted Zone ID',
            parameterName: 'tokillc-hostedzoneid',
            stringValue: domainZone.hostedZoneId,
        });

        new ssm.StringParameter(this, 'tokillc-hostedzonename', {
            description: 'Hosted Zone Name',
            parameterName: 'tokillc-hostedzonename',
            stringValue: domainZone.zoneName,
        });
    }
}