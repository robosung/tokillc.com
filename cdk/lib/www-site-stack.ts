/**
 * This file is the Stack for the actual random pet app itself. It declares the
 * needed resources which cdk-pipeline-stack.ts then deploys.
 * A starting version of this file is created by the cdk init command.
 *
 * Import required modules for CDK using ES6-style import directives per
 * https://docs.aws.amazon.com/cdk/latest/guide/work-with-cdk-typescript.html
 */
import {
  Duration,
  RemovalPolicy,
  Stack,
  StackProps
} from "aws-cdk-lib";
import { Construct } from "constructs";

import {
  aws_certificatemanager as acm,
  aws_cloudfront as cloudfront,
  aws_cloudfront_origins as cloudfrontorigin,
  aws_route53 as route53,
  aws_s3 as s3,
  aws_s3_deployment as s3deploy,
  aws_ssm as ssm,
  aws_route53_patterns as patterns
} from "aws-cdk-lib";

export class WebsiteStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps) {
    super(scope, id, props);

    const websiteBucketName = "www.tokillc.com";

    const certArn = ssm.StringParameter.valueForStringParameter(
      this,
      "tokillc-wildcardcertarn"
    );
    const zoneName = ssm.StringParameter.valueForStringParameter(
      this,
      "tokillc-hostedzonename"
    );
    const hostedZoneId = ssm.StringParameter.valueForStringParameter(
      this,
      "tokillc-hostedzoneid"
    );


    // Create S3 buckets to be used
    const websiteBucket = new s3.Bucket(this, "WebsiteBucket", {
      removalPolicy: RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
      encryption: s3.BucketEncryption.S3_MANAGED,
      publicReadAccess: false,
      bucketName: websiteBucketName,
    });
    // Deploy site contents to S3 bucket
    new s3deploy.BucketDeployment(this, "DeployWithInvalidation", {
      sources: [s3deploy.Source.asset("../src")],
      destinationBucket: websiteBucket,
      retainOnDelete: false,
    });

    const websiteCert = acm.Certificate.fromCertificateArn(
      this,
      "Certificate",
      certArn
    );

    const previewCachePolicy = new cloudfront.CachePolicy(
      this,
      "previewCachePolicy",
      {
        defaultTtl: Duration.minutes(30),
        minTtl: Duration.minutes(25),
        maxTtl: Duration.minutes(35),
        enableAcceptEncodingBrotli: true,
        enableAcceptEncodingGzip: true,
        headerBehavior:
          cloudfront.CacheHeaderBehavior.allowList("authorization"),
      }
    );

    const distribution = new cloudfront.Distribution(this, "SiteDistribution", {
      defaultBehavior: {
        origin: new cloudfrontorigin.S3Origin(websiteBucket),
        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        cachePolicy: previewCachePolicy,
        compress: true,
      },
      domainNames: [websiteBucketName],
      certificate: websiteCert,
      defaultRootObject: "index.html",
    });

    // DNS for the website using existing hosted zone
    const hostedZone = route53.HostedZone.fromHostedZoneAttributes(
      this,
      "Zone",
      {
        zoneName,
        hostedZoneId,
      }
    );

    new route53.CnameRecord(this, "DistributionCname", {
      zone: hostedZone,
      recordName: 'www',
      // Switch to CloudFlare DNS name
      domainName: distribution.distributionDomainName,
      ttl: Duration.seconds(300),
    });

    new patterns.HttpsRedirect(this, 'ApexRedirect', {
      recordNames: ['tokillc.com'], // Your apex domain
      targetDomain: 'www.tokillc.com', // The www subdomain
      zone: hostedZone,
    });

  }
}
