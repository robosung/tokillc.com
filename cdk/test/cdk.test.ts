import { Stack } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import * as customContext from "../bin/custom-context.json";
import { WebsiteStack } from "../lib/www-site-stack";
import { normalizeTemplate } from "@teamexos/infra-template-normalize";

const accountId = process.env.AWS_ACCT_ID ?? "482638059955";
const region = process.env.AWS_REGION ?? "us-east-1";
const accountConfig = customContext[accountId as keyof typeof customContext];

const corpAccountId = "192231937478";
describe("FontsWebsiteStack", () => {
  test("Snapshot Test", () => {
    const stack = new Stack();

    const env = {
      account: corpAccountId,
      region: region,
    };
    const fontwebsiteStack = new WebsiteStack(
      stack,
      "fontwebsiteStack",
      { env: env },
      "dev"
    );
    const template = Template.fromStack(fontwebsiteStack);
    const normalizedTemplate = normalizeTemplate(template);
    expect(normalizedTemplate).toMatchSnapshot();
  });
});
