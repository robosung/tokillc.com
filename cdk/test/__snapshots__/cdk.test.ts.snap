// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontsWebsiteStack Snapshot Test 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueexoshostedzoneidC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "exos-hostedzoneid",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueexoshostedzonenameC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "exos-hostedzonename",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueteamexoswildcardcertarnC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "teamexos-wildcardcertarn",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": {
      "DependsOn": [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-192231937478-us-east-1",
          "S3Key": "ANY",
        },
        "Environment": {
          "Variables": {
            "AWS_CA_BUNDLE": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
          },
        },
        "Handler": "index.handler",
        "Layers": [
          {
            "Ref": "DeployWithInvalidationAwsCliLayerDEDD5787",
          },
        ],
        "Role": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.9",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-192231937478-us-east-1",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-192231937478-us-east-1/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "WebsiteBucket75C24D94",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "WebsiteBucket75C24D94",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": [
          {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": {
      "DependsOn": [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-192231937478-us-east-1",
          "S3Key": "ANY",
        },
        "Description": {
          "Fn::Join": [
            "",
            [
              "Lambda function for auto-deleting objects in ",
              {
                "Ref": "WebsiteBucket75C24D94",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "index.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DeployWithInvalidationAwsCliLayerDEDD5787": {
      "Properties": {
        "Content": {
          "S3Bucket": "cdk-hnb659fds-assets-192231937478-us-east-1",
          "S3Key": "ANY",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "DeployWithInvalidationCustomResourceE3FF7455": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "DestinationBucketName": {
          "Ref": "WebsiteBucket75C24D94",
        },
        "Prune": true,
        "RetainOnDelete": false,
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": [
          "cdk-hnb659fds-assets-192231937478-us-east-1",
        ],
        "SourceObjectKeys": [
          "b39d38bd949b899fce09f2005f0e92982972649bfac9681ffae869244b4a44cb.zip",
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "DistributionCnameF6960C04": {
      "Properties": {
        "HostedZoneId": {
          "Ref": "SsmParameterValueexoshostedzoneidC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "Name": {
          "Fn::Join": [
            "",
            [
              "fonts-dev.",
              {
                "Ref": "SsmParameterValueexoshostedzonenameC96584B6F00A464EAD1953AFF4B05118Parameter",
              },
              ".",
            ],
          ],
        },
        "ResourceRecords": [
          "fonts-dev.teamexos.com.cdn.cloudflare.net",
        ],
        "TTL": "300",
        "Type": "CNAME",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "SiteDistribution3FF9535D": {
      "Properties": {
        "DistributionConfig": {
          "Aliases": [
            "fonts-dev.teamexos.com",
          ],
          "DefaultCacheBehavior": {
            "CachePolicyId": {
              "Ref": "previewCachePolicy0FC62D16",
            },
            "Compress": true,
            "TargetOriginId": "fontwebsiteStackSiteDistributionOrigin1952F1185",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": [
            {
              "DomainName": {
                "Fn::GetAtt": [
                  "WebsiteBucket75C24D94",
                  "RegionalDomainName",
                ],
              },
              "Id": "fontwebsiteStackSiteDistributionOrigin1952F1185",
              "S3OriginConfig": {
                "OriginAccessIdentity": {
                  "Fn::Join": [
                    "",
                    [
                      "origin-access-identity/cloudfront/",
                      {
                        "Ref": "SiteDistributionOrigin1S3Origin8B65D95B",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "ViewerCertificate": {
            "AcmCertificateArn": {
              "Ref": "SsmParameterValueteamexoswildcardcertarnC96584B6F00A464EAD1953AFF4B05118Parameter",
            },
            "MinimumProtocolVersion": "TLSv1.2_2021",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "SiteDistributionOrigin1S3Origin8B65D95B": {
      "Properties": {
        "CloudFrontOriginAccessIdentityConfig": {
          "Comment": "Identity for fontwebsiteStackSiteDistributionOrigin1952F1185",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "WebsiteBucket75C24D94": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "BucketName": "fonts-dev-teamexos.com",
        "Tags": [
          {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
          {
            "Key": "aws-cdk:cr-owned:b4b657d6",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteBucketAutoDeleteObjectsCustomResource8750E461": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "WebsiteBucketPolicyE10E3262",
      ],
      "Properties": {
        "BucketName": {
          "Ref": "WebsiteBucket75C24D94",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteBucketPolicyE10E3262": {
      "Properties": {
        "Bucket": {
          "Ref": "WebsiteBucket75C24D94",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:PutBucketPolicy",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "WebsiteBucket75C24D94",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "WebsiteBucket75C24D94",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": {
                "CanonicalUser": {
                  "Fn::GetAtt": [
                    "SiteDistributionOrigin1S3Origin8B65D95B",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "WebsiteBucket75C24D94",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "previewCachePolicy0FC62D16": {
      "Properties": {
        "CachePolicyConfig": {
          "DefaultTTL": 1800,
          "MaxTTL": 2100,
          "MinTTL": 1500,
          "Name": "fontwebsiteStackpreviewCachePolicyEADFE305-us-east-1",
          "ParametersInCacheKeyAndForwardedToOrigin": {
            "CookiesConfig": {
              "CookieBehavior": "none",
            },
            "EnableAcceptEncodingBrotli": true,
            "EnableAcceptEncodingGzip": true,
            "HeadersConfig": {
              "HeaderBehavior": "whitelist",
              "Headers": [
                "authorization",
              ],
            },
            "QueryStringsConfig": {
              "QueryStringBehavior": "none",
            },
          },
        },
      },
      "Type": "AWS::CloudFront::CachePolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
